azurelinuxagent/ga/logcollector.py

```diff
@@ -367,11 +371,15 @@ def collect_logs_and_get_archive(self):
                         archive_file_name = LogCollector._convert_file_name_to_archive_name(file_to_collect)
                         compressed_archive.write(file_to_collect.encode("utf-8"), arcname=archive_file_name)
                     except Exception as e:
-                        error_count += 1
-                        if error_count >= max_errors:
-                            raise Exception("Too many errors, giving up. Last error: {0}".format(ustr(e)))
+                        # [Errno 2] No such file or directory
+                        if isinstance(e, IOError) and e.errno == 2:     # pylint: disable=E1101
+                            _LOGGER.warning("File %s does not exist, skipping collection for this file", file_to_collect)
                         else:
-                            _LOGGER.warning("Failed to add file %s to the archive: %s", file_to_collect, ustr(e))
+                            error_count += 1
```

Comment:
keep existing error handling logic for other errors

Reply:
rather than 

`if isinstance(e, IOError)`

we should have 2 except clauses

```
except IOError as io_error:
    ...
except Exception as e:
   ...
```

That would also get rid of the no-member warning